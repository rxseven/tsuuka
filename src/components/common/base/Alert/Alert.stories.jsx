import React from 'react';
import { action } from '@storybook/addon-actions';
import { storiesOf } from '@storybook/react';

import Alert from './index';

const title = 'Components/Common/Base/Alert';
const props = { children: 'Alert message', onDismiss: action('onDismiss') };

storiesOf(title, module)
  .addParameters({
    info: {
      text: `
  **Usage:** for more information, see [Bootstrap - Alerts](https://getbootstrap.com/docs/4.1/components/alerts/).
`
    }
  })
  .add('danger (default)', () => <Alert {...props} />)
  .add('look', () => <Alert {...props} look="success" />);
