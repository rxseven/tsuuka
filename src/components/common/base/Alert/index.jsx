import cx from 'classnames';
import PropTypes from 'prop-types';
import exact from 'prop-types-exact';
import React from 'react';

const propTypes = exact({
  children: PropTypes.node.isRequired,
  look: PropTypes.string,
  onDismiss: PropTypes.func.isRequired
});

const defaultProps = {
  look: 'danger'
};

function Alert({ children, onDismiss, look }) {
  function handleDismiss() {
    onDismiss(children);
  }

  return (
    <div className={cx('alert', `alert-${look}`)}>
      {children}
      <button
        aria-label="Close"
        className="close"
        onClick={handleDismiss}
        type="button"
      >
        <span aria-hidden="true">&times;</span>
      </button>
    </div>
  );
}

Alert.propTypes = propTypes;
Alert.defaultProps = defaultProps;

export default Alert;
