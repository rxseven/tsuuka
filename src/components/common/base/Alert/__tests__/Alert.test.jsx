import { shallow } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';

import Alert from '../index';

describe('<Alert />', () => {
  // Arrange
  const props = { children: 'content', onDismiss: jest.fn() };
  const component = <Alert {...props} />;

  describe('Unit tests', () => {
    it('should render without crashing', () => {
      // Act
      const wrapper = shallow(component);

      // Assert
      expect(wrapper).toBeDefined();
    });
  });

  describe('Snapshot tests', () => {
    it('should render correctly', () => {
      // Act
      const tree = renderer.create(component).toJSON();

      // Assert
      expect(tree).toMatchSnapshot();
    });
  });
});
