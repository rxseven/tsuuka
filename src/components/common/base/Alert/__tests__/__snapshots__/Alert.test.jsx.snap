// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Alert /> Snapshot tests should render correctly 1`] = `
<div
  className="alert alert-danger"
>
  content
  <button
    aria-label="Close"
    className="close"
    onClick={[Function]}
    type="button"
  >
    <span
      aria-hidden="true"
    >
      Ã—
    </span>
  </button>
</div>
`;
